rust/src/format/metadata.rs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::format::pb;
+
+/// Data File Metadata
+#[derive(Debug, Default)]
+pub struct Metadata {
+    /// Offset of each record batch.
+    pub batch_offsets: Vec<i32>,
+
+    /// The file position of the page table in the file.
+    pub page_table_position: usize,
+
+    /// The file position of the manifest block in the file.
+    pub manifest_position: Option<usize>,
```

Comment:
Is the manifest block not required?

Reply:
So in "Dataset", there is one manifest file for the entire version of the dataset, each individual file does not need to have manifest.